dm/syncer/dml.go

```diff
@@ -686,14 +687,18 @@ func genKeyList(table string, columns []*model.ColumnInfo, dataSeq []interface{}
 }
 
 // truncateIndexValues truncate prefix index from data.
-func truncateIndexValues(indexColumns *model.IndexInfo, tiColumns []*model.ColumnInfo, data []interface{}) []interface{} {
+func truncateIndexValues(ctx sessionctx.Context, indexColumns *model.IndexInfo, tiColumns []*model.ColumnInfo, data []interface{}) []interface{} {
 	values := make([]interface{}, 0, len(indexColumns.Columns))
+	datums, err := utils.AdjustBinaryProtocolForDatum(ctx, data, tiColumns)
```

Comment:
seems we can use `tablecodec.TruncateIndexValues` to save a for loop below

Reply:
yes, `AdjustBinaryProtocolForDatum` which is diffrent from `types.NewDatum` returns `[]types.Datum` , i will fix.